src/box/box.cc

```diff
@@ -3100,6 +3100,70 @@ box_process1(struct request *request, box_tuple_t **result)
 	return box_process_rw(request, space, result);
 }
 
+void
+box_iterator_position_pack(const char *pos, const char *pos_end,
+			   uint32_t found, const char **packed_pos,
+			   const char **packed_pos_end)
+{
+	assert(packed_pos != NULL);
+	assert(packed_pos_end != NULL);
+	if (found > 0 && pos != NULL) {
+		uint32_t buf_size =
+			iterator_position_pack_bufsize(pos, pos_end);
+		char *buf =
+			(char *)xregion_alloc(&fiber()->gc, buf_size);
+		iterator_position_pack(pos, pos_end, buf, buf_size,
+				       packed_pos, packed_pos_end);
+	} else {
+		*packed_pos = NULL;
+		*packed_pos_end = NULL;
+	}
+}
+
+int
+box_iterator_position_unpack(const char *packed_pos,
+			     const char *packed_pos_end,
+			     struct key_def *cmp_def, const char *key,
+			     uint32_t key_part_count, int iterator,
+			     bool on_region, const char **pos,
+			     const char **pos_end, char **out_buf,
+			     uint32_t *out_buf_size)
+{
+	assert(pos != NULL);
+	assert(pos_end != NULL);
+	/* See NB section in the description of the function. */
+	assert(on_region || (out_buf != NULL && out_buf_size != NULL));
```

Comment:
We don't need `on_region` argument at all then - we can just do a runtime alloca and copy position there right in `box_index_iterator` instead of this strange argument.

Reply:
Right. Let's try that?

Reply:
Done, that's better.